tools/gen-intrinsics-specs.py

```diff
@@ -623,9 +626,78 @@ def get_section_data(row):
 
 
 def process_db(db, classification_db):
+    """Processes a list of intrinsics and their mappings to the
+    classification into a sequence of sections and RST tables.
+
+    >>> intrinsics = [
+    ... ['<HEADER>', 'T1', 'T2', 'T3', 'T4', 'T5'],
+    ... ['<SECTION>','Section 1 title', 'Section 1 description.'],
+    ... ['a A01()','a','aa','aaa','aaaa'],
+    ... ['b B01()','b','bb','bbb','bbbb'],
+    ... ['<SECTION>','Section 2 title', 'Section 2 description.'],
+    ... ['c C01()','c','cc','ccc','cccc'],
+    ... ]
+    >>> classification = {
+    ... 'B01': 'Section 1.1|Section 1.1.1',
+    ... 'C01': 'classX|subclassY'
+    ... }
+    >>> print(process_db(intrinsics, classification))
+    <BLANKLINE>
+    <BLANKLINE>
+    Section 1 title
+    ===============
+    <BLANKLINE>
+    Section 1 description.
+    <BLANKLINE>
+    No category
+    ~~~~~~~~~~~
+    <BLANKLINE>
+    +-------------+-------+--------+---------+----------+
+    | T1          | T2    | T3     | T4      | T5       |
+    +=============+=======+========+=========+==========+
+    | .. code:: c | ::    | ::     | ::      | ``aaaa`` |
+    |             |       |        |         |          |
+    |     a A01() |     a |     aa |     aaa |          |
```

Comment:
This test fails on Ubuntu 18.04. Instead of the nicely formatted table here, the output looks like this:
```
    +------+------+------+------+----------+
    | T1   | T2   | T3   | T4   | T5       |
    +======+======+======+======+==========+
    | .. code:: c
    <BLANKLINE>
        a A01()      | ::
    <BLANKLINE>
        a      | ::
    <BLANKLINE>
        aa      | ::
    <BLANKLINE>
        aaa      | ``aaaa`` |
    +------+------+------+------+----------+
```
It looks to me as if that's because 18.04's version of the `python3-tabulate` package isn't up-to-date enough to support multi-line records in tables.

It runs as expected on Ubuntu 20.04, though. (Between those releases, `python3-tabulate` went from 0.7.7 to 0.8.6.)

I suspect there's no good way to make that work, so perhaps checking the version of `tabulate` might be a simple answer, so that (again) we can generate a sensible error message?

Reply:
I have added the check. Thank you for the review!